pkg/worker/verifier/latest_value_map.go

```diff
@@ -0,0 +1,97 @@
+package verifier
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/redpanda-data/kgo-verifier/pkg/util"
+	log "github.com/sirupsen/logrus"
+)
+
+type LatestValueMap struct {
+	topic       string
+	latestValue []map[string]string
+}
+
+func (lvm *LatestValueMap) Get(partition int32, key string) (value string, exists bool) {
+	if partition < 0 || partition >= int32(len(lvm.latestValue)) {
+		return "", false
+	}
+	value, exists = lvm.latestValue[partition][key]
+	if exists {
+		return value, true
+	}
+	return "", false
+}
+
+func (lvm *LatestValueMap) Insert(partition int32, key string, value string) {
+	lvm.latestValue[partition][key] = value
+}
+
+func latestValueMapFile(topic string) string {
+	return fmt.Sprintf("latest_value_%s.json", topic)
+}
+
+func (lvm *LatestValueMap) Store() error {
+	log.Infof("LatestValueMap::Storing %s", latestValueMapFile(lvm.topic))
+
+	data, err := json.Marshal(lvm)
+	if err != nil {
+		return err
+	}
+
+	tmp_file, err := ioutil.TempFile("./", "latest_value_*.tmp")
+	if err != nil {
+		return err
+	}
+
+	_, err = tmp_file.Write(data)
+	if err != nil {
+		return err
+	}
+
+	err = os.Rename(tmp_file.Name(), latestValueMapFile(lvm.topic))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func LoadLatestValues(topic string, nPartitions int32) LatestValueMap {
+	data, err := ioutil.ReadFile(latestValueMapFile(topic))
+	if err != nil {
+		// Pass, assume it's not existing yet
+		log.Warnf("Can't read topic latest value map: %v", err)
```

Comment:
I find this very error prone. We should have a command line parameter which explicitly specifies whether the latest values must be present or not otherwise we might get false positives where the test passes but only because we didn't put the file on the disk properly...

Reply:
Done